### PR TITLE
minc2-simple: init at 2018-12-20

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Motivation for this change

Adds the C and Python parts of the minc2-simple library (and one associated script) in separate expressions.

The tests for the C library seem to be broken but the Python tests succeed.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [NA] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [NA] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by bcdarwin** [comment]
Looks like this PR was initially ignored due to a merge conflict?  Anyway, fixed, patches merged upstream, bumped to latest Git commit.

@FRidh

File: pkgs/development/libraries/libminc/default.nix [hunk 1]
```diff
@@ -19,7 +19,7 @@ stdenv.mkDerivation rec {
   '';
 
   nativeBuildInputs = [ cmake ];
-  buildInputs = [ zlib netcdf nifticlib hdf5 ];
+  propagatedBuildInputs = [ zlib netcdf nifticlib hdf5 ];
```
Comment (FRidh) [inline pkgs/development/libraries/libminc/default.nix]: Avoid propagating libraries.

File: pkgs/development/libraries/minc2-simple/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchFromGitHub, cmake, libminc }:
+
+stdenv.mkDerivation rec {
+  pname    = "minc2-simple";
+  name     = "${pname}-2018-12-20";
```
Comment (FRidh) [inline pkgs/development/libraries/minc2-simple/default.nix]: drop name, instead set version

File: pkgs/development/libraries/minc2-simple/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  src = fetchFromGitHub {
+    owner  = "vfonov";
+    repo   = pname;
+    rev    = "e04414eb74052d8a6f9fb4bedda7f0894e86ac82";
+    sha256 = "1miy942ibbcikg0k45d7342ags2gpr29rvhvspwxqgv2jnpcwg05";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  propagatedBuildInputs = [ libminc ];
```
Comment (FRidh) [inline pkgs/development/libraries/minc2-simple/default.nix]: why propagate?

File: pkgs/development/python-modules/minc2-simple/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{ buildPythonPackage, fetchFromGitHub, stdenv, cmake, minc2_simple,
+  cffi, six, numpy, scipy, pytest, minc_tools, mni_autoreg }:
+
+buildPythonPackage rec {
+  pname    = "minc2-simple";
+  name     = "${pname}-2018-12-20";
```
Comment (FRidh) [inline pkgs/development/python-modules/minc2-simple/default.nix]: drop name, set version

File: pkgs/development/python-modules/minc2-simple/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    rev    = "e04414eb74052d8a6f9fb4bedda7f0894e86ac82";
+    sha256 = "1miy942ibbcikg0k45d7342ags2gpr29rvhvspwxqgv2jnpcwg05";
+  };
+
+  checkInputs = [ pytest minc_tools mni_autoreg ];
+  propagatedBuildInputs = [ cffi six numpy scipy minc2_simple ];
+
+  preConfigure = "cd python";
+
+  testPhase = "pytest test";
```
Comment (FRidh) [inline pkgs/development/python-modules/minc2-simple/default.nix]: testPhase does not exist, it's `checkPhase`

File: pkgs/development/python-modules/minc2-simple/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/vfonov/${pname}";
```
Comment (FRidh) [inline pkgs/development/python-modules/minc2-simple/default.nix]: remove substitution here. It doesn't add anything.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/libminc/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/libraries/libminc/default.nix]: All propagatedBuildInputs removed.

File: pkgs/development/libraries/minc2-simple/default.nix (hunk 3 above)
Comment (bcdarwin) [reply pkgs/development/libraries/minc2-simple/default.nix]: Same.

File: pkgs/development/python-modules/minc2-simple/default.nix (hunk 5 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/minc2-simple/default.nix]: Removed.

File: pkgs/development/python-modules/minc2-simple/default.nix (hunk 6 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/minc2-simple/default.nix]: Done.

File: pkgs/development/libraries/minc2-simple/default.nix (hunk 2 above)
Comment (bcdarwin) [reply pkgs/development/libraries/minc2-simple/default.nix]: Done.

File: pkgs/development/python-modules/minc2-simple/default.nix (hunk 4 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/minc2-simple/default.nix]: Done.

**Comment by bcdarwin** [comment]
Everything resolved.

**Comment by bcdarwin** [comment]
@Fridh this has picked up a trivial merge conflict from sitting around but otherwise should be fine to merge.

**Pull request closed without merging**